src/Microsoft.Extensions.ProjectModel.Sources/Resolution/DependencyDescription.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.ProjectModel.Resolution
+{
+    public class DependencyDescription
+    {
+        private const string DefaultVersion = "1.0.0";
+        private List<Dependency> _dependencies;
+
+        public DependencyDescription(string name, string version, string path, string framework, string type, bool isResolved)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if(string.IsNullOrEmpty(framework))
+            {
+                throw new ArgumentNullException(nameof(framework));
+            }
+
+            Name = name;
+            Version = string.IsNullOrEmpty(version) ? DefaultVersion : version;
+            TargetFramework = framework;
+            Resolved = isResolved;
+            Path = path;
+            DependencyType dt;
+            Type = Enum.TryParse(type, out dt) ? dt : DependencyType.Unknown;
```

Comment:
Can you make this case insensitive? `Enum.TryParse(type, ignoreCase: true, out dt)`